mei/mixins.py

```diff
@@ -170,21 +174,26 @@ class MEIMethodMixin:
         "postprocessing",
     )
 
-    def add_method(self, method_fn: str, method_config: Mapping, comment: str = "") -> None:
+    def add_method(self, method_fn: str, method_config: Mapping, comment: str = "", **kwargs) -> None:
         self.insert1(
             dict(
                 method_fn=method_fn,
                 method_hash=make_hash(method_config),
                 method_config=method_config,
                 method_comment=comment,
-            )
+            ),
+            **kwargs,
         )
 
     def generate_mei(self, dataloaders: Dataloaders, model: Module, key: Key, seed: int) -> Dict[str, Any]:
         method_fn, method_config = (self & key).fetch1("method_fn", "method_config")
         method_fn = self.import_func(method_fn)
         self.insert_key_in_ops(method_config=method_config, key=key)
-        mei, score, output = method_fn(dataloaders, model, method_config, seed)
+
+        train_input_shape = next(iter(next(iter(dataloaders["train"].values())))).images.shape
+        input_shape = (1, 4) + train_input_shape[2:]
```

Comment:
But I have this comment: Why is this here? I guess batch size for MEI should be one but why is the 4 there?

Reply:
the 4 are the behavior channels (for behavior we append 3 behavior variables to 1 color channel). These are the input batches returned by the dataloader and expected by the model. It should not be hardcoded though, I fixed it

Reply:
Great, thanks. I think this is better because for example in the retina, there is no behavior